### PR TITLE
test(react): add test for ErrorBoundary to reject empty array for shouldCatch prop

### PR DESCRIPTION
**Pull request by manudeli**

### CODE REVIEW

File: packages/react/src/ErrorBoundary.tsx [hunk 1]
```diff
@@ -40,72 +40,50 @@ export interface ErrorBoundaryFallbackProps<TError extends Error = Error> extend
...
 
-/**
- * Type guard function that narrows error type
- */
-type ErrorTypeGuard<TError extends Error = Error> = (error: Error) => error is TError
-
-/**
- * Validation function without type narrowing
- */
+type ErrorTypeGuard<TError extends Error> = (error: Error) => error is TError
```
Comment (manudeli) [inline packages/react/src/ErrorBoundary.tsx:43]: I thought ErrorTypeGuard is already good naming. so I think we can remove jsdoc

File: packages/react/src/ErrorBoundary.tsx [hunk 2]
```diff
@@ -40,72 +40,50 @@ export interface ErrorBoundaryFallbackProps<TError extends Error = Error> extend
...
 type ErrorValidator = (error: Error) => boolean
```
Comment (manudeli) [inline packages/react/src/ErrorBoundary.tsx:44]: I thought ErrorValidator is already good naming. so I think we can remove jsdoc

File: packages/react/src/ErrorBoundary.tsx [hunk 3]
```diff
@@ -40,72 +40,50 @@ export interface ErrorBoundaryFallbackProps<TError extends Error = Error> extend
...
 
-type ErrorMatcher<TError extends Error = Error> =
-  | boolean
-  | ConstructorType<TError>
-  | ErrorTypeGuard<TError>
-  | ErrorValidator
-
-type ShouldCatch<TError extends Error = Error> = ErrorMatcher<TError> | readonly ErrorMatcher<TError>[]
+type ErrorMatcher = boolean | ConstructorType<Error> | ErrorTypeGuard<Error> | ErrorValidator
```
Comment (manudeli) [inline packages/react/src/ErrorBoundary.tsx:46]: IMO, ErrorMatcher shouldn't be generic

File: packages/react/src/ErrorBoundary.tsx [hunk 4]
```diff
@@ -185,7 +163,11 @@ class BaseErrorBoundary<TShouldCatch extends ShouldCatch = ShouldCatch> extends
       if (error instanceof ErrorInFallback) {
         throw error.originalError
       }
-      if (!shouldCatchError<InferError<TShouldCatch>>(shouldCatch as ShouldCatch<InferError<TShouldCatch>>, error)) {
```
Comment (manudeli) [inline packages/react/src/ErrorBoundary.tsx:188]: I removed unnecessary type assertion and generics
